### PR TITLE
fix(datepicker): call setDate when value is updated

### PR DESCRIPTION
**Pull request by tay1orjones**
[Surfaced in slack](https://ibm-studios.slack.com/archives/C2K6RFJ1G/p1655131902714149), the DatePicker wasn't setting the `value` when it was passed one. Small error using flatpickr's `set()` vs the intended `setDate()`

#### Changelog

**New**

- tests covering the use case of a controlled date picker

**Changed**

- Fix DatePicker to set `value` for flatpickr when given one

#### Testing / Reviewing

- I made a 'ForTesting' story that you can see is a simple example of controlling the input by tracking it's state change via onChange and passing new state via the 'clear' button. This should work as you would expect
- All other DatePicker stories should work as intended

Do not merge until I've removed the ForTesting story

### CODE REVIEW

File: packages/react/src/components/DatePicker/DatePicker-test.js [hunk 1]
```diff
@@ -225,4 +227,47 @@ describe('DatePicker', () => {
...
+        </>
+      );
+    };
+
+    render(<DatePickerExample />);
+    expect(screen.getByLabelText('Date Picker label')).toHaveValue('');
+
+    userEvent.type(
+      screen.getByLabelText('Date Picker label'),
+      '01/20/1989{enter}'
```
Comment (jnm2377) [inline packages/react/src/components/DatePicker/DatePicker-test.js:264]: woah, this is cool! never tried this one before!

**Review by jnm2377 — APPROVED** [review]
It's working as expected! 🔥

File: packages/react/src/components/DatePicker/DatePicker-test.js (hunk 1 above)
Comment (jnm2377) [reply packages/react/src/components/DatePicker/DatePicker-test.js:264]: Thanks for adding these tests!!

**Review by tw15egan — APPROVED** [review]
LGTM 👍 ✅

**Review by abbeyhrt — APPROVED** [review]

**Pull request merged**